remove-eb-migration-transient.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ * Plugin Name: Eventbrite Tickets Extension: Clean Up Migration Data
+ * Description: Older versions of Eventbrite Tickets would sometimes add a transient with no expiration date. This extension removes that transient if it exists.
+ * Version: 1.0.0
+ * Author: Modern Tribe, Inc.
+ * Author URI: http://m.tri.be/1971
+ * License: GPLv2 or later
+ */
+
+defined( 'WPINC' ) or die;
+
+class Tribe__Extension__Remove_Eventbrite_Migration_Transient {
+
+    /**
+     * The name of the migration transient that we may be deleting.
+     *
+     * @since 1.0.0
+     *
+     * @var string
+     */
+    public static $transient_key = '_tribe_eventbrite_is_migrating';
+
+    /**
+     * The name of the option where we store a simple boolean if the old transient was deleted.
+     *
+     * @since 1.0.0
+     *
+     * @var string
+     */
+    public static $deleted_key = '_tribe_deleted_old_migration_transient';
+
+    /**
+     * The path to this extension's main plugin file.
+     *
+     * @since 1.0.0
+     *
+     * @var string
+     */
+    public $extension_path = '';
+
+    /**
+     * Extension version.
+     *
+     * @var string
+     */
+    const VERSION = '1.0.0';
+
+    /**
+     * Required plugins as defined by their Main classes.
+     *
+     * @since 1.0.0
+     *
+     * @var string
+     */
+    public $plugins_required = array(
+        'Tribe__Events__Main'                      => '4.7',
+        'Tribe__Events__Tickets__Eventbrite__Main' => '4.5'
+    );
+
+    /**
+     * Ensures extension hooks run well after main plugin hooks.
+     *
+     * @since 1.0.0
+     */
+    public function __construct() {
+
+        $this->extension_path = plugin_basename( __FILE__ );
+
+        add_action( 'plugins_loaded', array( $this, 'init' ), 100 );
+    }
+
+    /**
+     * Simple check for whether the extension's okay to run.
+     *
+     * @since 1.0.0
+     *
+     * @return boolean
+     */
+    public function extension_should_run() {
+        return function_exists( 'tribe_register_plugin' ) && tribe_register_plugin( __FILE__, __CLASS__, self::VERSION, $this->plugins_required );
+    }
+
+    /**
+     * Extension hooks.
+     *
+     * @since 1.0.0
+     */
+    public function init() {
+
+        if ( ! $this->extension_should_run() ) {
+            return false;
+        }
+
+        add_action( 'init', array( $this, 'maybe_delete_old_transient' ) );
+        add_action( 'after_plugin_row_' . $this->extension_path, array( $this, 'maybe_add_deactivation_notice' ), 10, 3 );
+        add_action( 'admin_head', array( $this, 'admin_head_css' ) );
+    }
+
+    /**
+     * If the migration transient exists already, *and* doesn't have an expiration.
+     *
+     * @since 1.0.0
+     *
+     * @return boolean
+     */
+    public function old_transient_exists() {
+        $transient         = get_transient( self::$transient_key );
+        $transient_timeout = get_option( '_transient_timeout_' . self::$transient_key );
```

Comment:
On line 111 you're checking if `false === $transient_timeout`, but if the timeout is `0`, `''` or `null` that check would pass.
I suggest to cast the `get_option` return value to `bool`:
```php
$transient_timeout = (bool)get_option( '_transient_timeout_' . self::$transient_key );
```

Reply:
Hey Luca, my thinking here was this:

* That `_transient_timeout_` option only even gets _created_ if the `$timeout` from `set_transient` is **not** `0`.
* So I'm not really trying to check the value of the option, I'm checking if the option _exists_. If it doesn't, `false` will always be returned there.

Thoughts?